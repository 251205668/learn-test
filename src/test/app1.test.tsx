import renderer from "react-test-renderer";
import App1 from "@/test/app1";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

test("test Onlick", () => {
  const component = renderer.create(
    <App1 message="http://antfu.me">Anthony Fu</App1>
  );

  let tree = toJson(component);
  expect(tree).toMatchSnapshot();

  // trigger onlick
  // tree.props.

  // // re-rendering
  // tree = toJson(component);
  // expect(tree).toMatchSnapshot();
  // tree.props.onClick();
});
