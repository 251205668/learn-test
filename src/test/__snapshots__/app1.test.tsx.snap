// Vitest Snapshot v1

exports[`test Onlick 1`] = `
<div>
  <div>
    http://antfu.me
  </div>
  <div>
    Anthony Fu
  </div>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    click
  </button>
</div>
`;

exports[`test Onlick 2`] = `
<div>
  <div>
    http://antfu.me
  </div>
  <div>
    Anthony Fu
  </div>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    click
  </button>
</div>
`;
